### ISSUE TITLE
allow export fields to be customised on a DataObject

### ISSUE THREAD

**Issue by sunnysideup** [silverstripe/silverstripe-admin#1106]
I am looking at:

https://github.com/silverstripe/silverstripe-admin/blob/1/code/ModelAdmin.php#L303-L306

and I am wondering if we can do something like:

```php

if($model->hasMethod('getExportColumns')) {
    return model->getExportColumns();
}
```
or similar (e.g. private static setting) as this would make it much easier to export.

Also, I reckon for an export you want to include all DB and HAS_ONE fields by default because it is easy to delete columns afterwards, but not easy to add columns.

**Comment by kinglozzer** [comment]
Yeah I agree this would be handy, I often end up adding this to my `ModelAdmin` classes:

```php
    public function getExportFields()
    {
        $model = singleton($this->modelClass);
        if ($model->hasMethod('getExportFields')) {
            return model->getExportFields();
        }

        return singleton($this->modelClass)->summaryFields();
    }
```

We already do something similar for `getCMSValidator()` (“method exists” checks). We could either do that, or add a stub `DataObject::getExportFields()` method which mimics the existing behaviour of returning summary fields and call that.

Fancy submitting a PR?
